requirements.txt

```diff
@@ -1,130 +1,66 @@
-#
-# This file is autogenerated by pip-compile with Python 3.8
-# by the following command:
-#
-#    pip-compile
-#
 aiohttp==3.8.4
-    # via web3
 aiosignal==1.3.1
-    # via aiohttp
+astroid==3.0.3
 async-timeout==4.0.2
-    # via aiohttp
 attrs==22.2.0
-    # via
-    #   aiohttp
-    #   jsonschema
+base58==2.1.1
 bitarray==2.7.3
-    # via eth-account
 certifi==2022.12.7
-    # via requests
 charset-normalizer==3.1.0
-    # via
-    #   aiohttp
-    #   requests
+colorama==0.4.6
 cytoolz==0.12.1
-    # via eth-utils
+dill==0.3.8
 ecdsa==0.18.0
-    # via brine-connector (pyproject.toml)
-eth-abi==4.0.0b2
-    # via
-    #   eth-account
-    #   web3
-eth-account==0.8.0
-    # via web3
-eth-hash[pycryptodome]==0.5.1
-    # via
-    #   eth-utils
-    #   web3
-eth-keyfile==0.6.1
-    # via eth-account
-eth-keys==0.4.0
-    # via
-    #   eth-account
-    #   eth-keyfile
-eth-rlp==0.3.0
-    # via eth-account
-eth-typing==3.3.0
-    # via
-    #   eth-abi
-    #   eth-keys
-    #   eth-utils
-    #   web3
-eth-utils==2.1.0
-    # via
-    #   eth-abi
-    #   eth-account
-    #   eth-keyfile
-    #   eth-keys
-    #   eth-rlp
-    #   rlp
-    #   web3
+eth-abi==2.2.0
+eth-account==0.5.9
+eth-hash==0.5.1
+eth-keyfile==0.5.1
+eth-keys==0.3.4
+eth-rlp==0.2.1
+eth-typing==2.3.0
+eth-utils==1.9.5
+ethers==0.1.1
+exceptiongroup==1.2.0
 frozenlist==1.3.3
-    # via
-    #   aiohttp
-    #   aiosignal
 hexbytes==0.3.0
-    # via
-    #   eth-account
-    #   eth-rlp
-    #   web3
 idna==3.4
-    # via
-    #   requests
-    #   yarl
 importlib-resources==5.12.0
-    # via jsonschema
-jsonschema==4.17.3
-    # via web3
+iniconfig==2.0.0
+ipfshttpclient==0.8.0a2
+isort==5.13.2
+jsonschema==3.2.0
 lru-dict==1.1.8
-    # via web3
+mccabe==0.7.0
 mpmath==1.3.0
-    # via sympy
+multiaddr==0.0.9
 multidict==6.0.4
-    # via
-    #   aiohttp
-    #   yarl
-parsimonious==0.9.0
-    # via
-    #   eth-abi
-    #   web3
-pkgutil-resolve-name==1.3.10
-    # via jsonschema
-protobuf==4.22.1
-    # via web3
+netaddr==1.2.1
+packaging==23.2
+parsimonious==0.8.1
+pkgutil_resolve_name==1.3.10
+platformdirs==4.2.0
+pluggy==1.4.0
+protobuf==3.20.3
 pycryptodome==3.17
-    # via
-    #   eth-hash
-    #   eth-keyfile
+pylint==3.0.3
 pyrsistent==0.19.3
-    # via jsonschema
+pytest==8.0.0
+python-dotenv==1.0.1
```

Comment:
Can you generate the requirement files properly.
There are dependencies here which aren't required to be installed by users.
Start with dependencies and dev-dependencies in pyproject.toml and install them in separate virtual environments. And then use pip tools to create the requirement files.